RELEASE_NOTES.md

```diff
@@ -14,6 +14,11 @@
 - feat: New behavior for `import opened M` where `M` contains a top-level declaration `M`, see PR for a full description (https://github.com/dafny-lang/dafny/pull/2355)
 - fix: The DafnyServer package is now published to NuGet as well, which fixes the previously-broken version of the DafnyLanguageServer package. (https://github.com/dafny-lang/dafny/pull/2787)
 - deprecate: Statement-level refinement syntax (e.g. `assert ...`) is deprecated (https://github.com/dafny-lang/dafny/pull/2756)
+- docs: The user documentation at https://dafny.org has a new landing page
+and is now synchronized with 
```

Comment:
Fine with releasing this now, but we need to quickly follow up with a change to `docs/dev/RELEASE.md` or else this is just a good intention. :)